### PR TITLE
Maybe fix memory leak caused by not disposing Icons and Bitmaps.

### PR DESCRIPTION
**Pull request by Niko-O**
Icon implements IDisposable and must therefore be disposed.
AudioDeviceIconExtractor creates and caches a lot of icons but never
frees old cache entries. TrayIcon.UpdateIcon also creates many copies of
Resources.SoundSwitch16. Getters in the generated Resource class always
return new instances of Bitmaps and Icons.
This commit removes the caching and adds code to manually dispose icons
used in various locations. Some uses of the Resource class are cached
instead (where the resource is not disposed anyways) to avoid loading
the same resource multiple times from disk. I didn't touch generated
files, though.

This commit is untested since I cannot compile the project.

### CODE REVIEW

**Comment by Niko-O** [comment]
Leaving Icons laying around eventually triggers an ExternalException when calling Bitmap.GetHicon, caused by GDI+ not wanting to give you any more handles. This has happened to a friend, who has asked me to maybe try and fix it.
While I was at it, and since it overlaps in one specific instance, I have changed all uses of the generated Resource class (except in generated code). Getters in this class always load the resource from disk. I have added rudimentary code to cache the returned objects.

Since I cannot compile the project (Visual Studio can't find some of the referenced assemblies, like NAudio) I cannot test this fix. Please make sure to test everything before merging.
There may be some cases I have missed, so the issue might not be completely fixed.

So in short:
Using a getter in the generated Resource class creates a new object.
Calling Bitmap.GetHicon creates a new unmanaged resource, which is automatically freed when the Icon created by Icon.FromHandle with the returned handle is disposed.
Calling Icon.ToBitmap creates a new object.
And it might even be the case that the setter of NotifyIcon.Icon internally creates a new object too. This does happen with the setter of Form.Icon.

**Comment by Belphemur** [comment]
Hello Niko,

Thanks for this, I see the problem. 

I've merged your changes, but with keeping the cache, only a different implementation, an expirable one. If the Icon is not retrieved for 5 minutes, it's disposed and removed from the cache.

This should resolve any issue there were with the icons.

**Comment by Belphemur** [comment]
Also about NAudio, you need to fetch the NuGet package for the project to build.

**Comment by Niko-O** [comment]
Great to hear that. Caching icons for 5 minutes should work fine.
If you have any questions about my pull request or if there are any problems with it, don't hesitate to contact me.
